### PR TITLE
Fix undefined default_config in configuration_utils.py

### LINKED ISSUE #33927: [Bug] Error when saving NVLM model
### System Info

```
- `transformers` version: 4.45.1
- Platform: Linux-6.5.0-35-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.23.5
- Safetensors version: 0.4.3
- Accelerate version: 0.34.2
- Accelerate config:    not found
- PyTorch version (GPU?): 2.4.0+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: Just using `device_map="auto"`
- Using GPU in script?: H100
- GPU type: NVIDIA H100 80GB HBM3
```

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The error is due to an undefined variable:
```
  File "/home/mgoin/venvs/test/lib/python3.10/site-packages/transformers/modeling_utils.py", line 2616, in save_pretrained
    misplaced_generation_parameters = model_to_save.config._get_non_default_generation_parameters()
  File "/home/mgoin/venvs/test/lib/python3.10/site-packages/transformers/configuration_utils.py", line 1050, in _get_non_default_generation_parameters
    if default_config is not None:
UnboundLocalError: local variable 'default_config' referenced before assignment
```

Looking at where that variable would be assigned, there is a case where decoder_config is set to None, but not default_config
```python
        # Composite models don't have a default config, use their decoder config as a fallback for default values
        # If no known pattern is matched, then `default_config = None` -> check against the global generation defaults
        try:
            default_config = self.__class__()
        except ValueError:
            decoder_config = self.get_text_config(decoder=True)
            if decoder_config is not self:
                default_config = decoder_config.__class__()
            else:
                decoder_config = None
```

### Expected behavior

Saving the model should not result in a crash

### PR DESCRIPTION
**Pull request by mgoin**
# What does this PR do?

Fixes https://github.com/huggingface/transformers/issues/33927

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
cc @gante

File: src/transformers/configuration_utils.py [hunk 1]
```diff
@@ -1036,6 +1036,7 @@ def _get_non_default_generation_parameters(self) -> Dict[str, Any]:
             if decoder_config is not self:
                 default_config = decoder_config.__class__()
             else:
+                default_config = None
                 decoder_config = None
```
Comment (gante) [inline src/transformers/configuration_utils.py:1040]:
```suggestion
                default_config = None
```

Comment (gante) [reply src/transformers/configuration_utils.py:1040]: it was a typo -- we don't need to set `decoder_config` 👀

**Review by gante — APPROVED** [review]
Good catch! Thank you for fixing 💛

File: src/transformers/configuration_utils.py (hunk 1 above)
Comment (SunMarc) [inline src/transformers/configuration_utils.py:1040]: I think that there was a typo in this [PR](https://github.com/huggingface/transformers/pull/33219/files) and we should only set `default_config = None` and remove `decoder_config = None` cc @gante

**Review by SunMarc — APPROVED** [review]
Make sense ! I left a comment

**Review by LysandreJik — APPROVED** [review]

**Pull request merged**